app/src/main/java/net/squanchy/home/HomeActivity.java

```diff
@@ -177,12 +183,31 @@ protected void onResume() {
     }
 
     private void handleProximityEvent(ProximityEvent proximityEvent) {
-        // TODO highlight speech near the rooms
-        if (proximityEvent.action().equals(KEY_CONTEST_STAND)) {
-            navigator.toContest(proximityEvent.subject());
+        switch (proximityEvent.action()) {
+            case KEY_CONTEST_STAND:
+                navigator.toContest(proximityEvent.subject());
+                break;
+            case KEY_ROOM_EVENT:
+                showCurrentEvent(proximityEvent.subject());
+                break;
         }
     }
 
+    private void showCurrentEvent(String placeId) {
+        currentEventSnackbarService.eventIn(placeId)
+                .map(this::toSnackbar)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(Snackbar::show);
+    }
+
+    private Snackbar toSnackbar(Event event) {
+        return currentEventSnackbarService.buildSnackbar(
```

Comment:
:x: As mentioned in the service, let's do the whole snackbar creation here, not there